webapp/src/main/java/life/qbic/views/register/RegisterHandler.java

```diff
@@ -3,71 +3,90 @@
 import com.vaadin.flow.component.Key;
 import com.vaadin.flow.component.UI;
 import life.qbic.usermanagement.User;
-import life.qbic.usermanagement.persistence.UserJpaRepository;
+import life.qbic.usermanagement.User.UserException;
+import life.qbic.usermanagement.registration.RegisterUserInput;
+import life.qbic.usermanagement.registration.RegisterUserOutput;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-
 /**
  * Handles the {@link RegisterLayout} components
  *
  * <p>This class is responsible for enabling buttons or triggering other view relevant changes on
  * the view class components
  */
 @Component
-public class RegisterHandler implements RegisterHandlerInterface {
+public class RegisterHandler implements RegisterHandlerInterface, RegisterUserOutput {
+
   private static final org.apache.logging.log4j.Logger log =
       org.apache.logging.log4j.LogManager.getLogger(RegisterHandler.class);
+  private RegisterLayout userRegistrationLayout;
 
-  private final UserJpaRepository userJpaRepository;
-  private RegisterLayout registeredRegisterLayout;
+  private final RegisterUserInput registerUserInput;
 
-  RegisterHandler(UserJpaRepository userJpaRepository) {
-    this.userJpaRepository = userJpaRepository;
+  @Autowired
+  RegisterHandler(RegisterUserInput registerUserInput) {
+    this.registerUserInput = registerUserInput;
+    registerUserInput.setOutput(this);
   }
 
   @Override
   public boolean register(RegisterLayout registerLayout) {
-    if (registeredRegisterLayout != registerLayout) {
-      this.registeredRegisterLayout = registerLayout;
+    if (userRegistrationLayout != registerLayout) {
+      this.userRegistrationLayout = registerLayout;
```

Comment:
I think by this renaming its not clear what the difference between the two layouts is. One was already registered in the layout, the other not

Reply:
I am confused with too many `registered` here. I think I will do another round of renaming to make it more explicit.